cli/compat/esm_resolver.rs

```diff
@@ -121,9 +129,23 @@ fn node_resolve(
   let conditions = DEFAULT_CONDITIONS;
   let url = module_resolve(specifier, &parent_url, conditions)?;
 
+  let resolve_response = if url.as_str().starts_with("http") {
+    ResolveResponse::Esm(url)
+  } else if url.as_str().ends_with(".js") {
```

Comment:
We should be doing case insensitive comparisons. It seems like we often forget about this. Maybe we should create a helper function somewhere that does it efficiently and just in case, also checks `url.path()` instead of the full url?

Reply:
@bartlomieju same again